homeassistant/components/ollama/config_flow.py

```diff
@@ -0,0 +1,245 @@
+                return self.async_show_progress_done(next_step_id="failed")
+
+            return self.async_show_progress_done(next_step_id="finish")
+
+        return self.async_show_progress(
+            step_id="download",
+            progress_action="download",
+            progress_task=self.download_task,
+        )
+
+    async def async_step_finish(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step after model downloading has succeeded."""
+        assert self.url is not None
+        assert self.model is not None
+
+        return self.async_create_entry(
+            title=_get_title(self.model),
+            data={CONF_URL: self.url, CONF_MODEL: self.model},
+        )
+
+    async def async_step_failed(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step after model downloading has failed."""
+        return self.async_abort(reason="download_failed")
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return OllamaOptionsFlow(config_entry)
+
+
+class OllamaOptionsFlow(OptionsFlow):
+    """Ollama options flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.url: str = self.config_entry.data[CONF_URL]
+        self.model: str = self.config_entry.data[CONF_MODEL]
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=_get_title(self.model), data=user_input
+            )
+
+        options = self.config_entry.options or MappingProxyType({})
+        schema = ollama_config_option_schema(options)
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(schema),
+        )
+
+
+def ollama_config_option_schema(options: MappingProxyType[str, Any]) -> dict:
+    """Ollama options schema."""
+    return {
+        vol.Optional(
+            CONF_PROMPT,
+            description={"suggested_value": options.get(CONF_PROMPT, DEFAULT_PROMPT)},
+        ): TemplateSelector(),
+        vol.Optional(
+            CONF_MAX_HISTORY,
+            description={
+                "suggested_value": options.get(CONF_MAX_HISTORY, DEFAULT_MAX_HISTORY)
+            },
+        ): NumberSelector(
+            NumberSelectorConfig(
+                min=0, max=sys.maxsize, step=1, mode=NumberSelectorMode.BOX
+            )
+        ),
+    }
+
+
+def _get_title(model: str) -> str:
+    """Get title for config entry."""
+    if model.endswith(":latest"):
+        model = model.split(":", maxsplit=1)[0]
+
+    return f"Ollama {model}"
```

Comment:
No need to prefix with Ollama. The integration is called Ollama, so it would end up with `Ollama: Ollama llama2`.
```suggestion
    return model
```